lib/src/run.dart

```diff
@@ -144,3 +165,66 @@ stderr:
 
 $stderr""";
 }
+
+
+/// Arguments passed to [Process.run] or [Process.start].
+/// See [Process.run] for more details.
+class RunOptions {
+  final String workingDirectory;
+  final Map<String, String> environment;
+  final bool includeParentEnvironment;
+  final bool runInShell;
+  final Encoding stdoutEncoding;
+  final Encoding stderrEncoding;
+
+  RunOptions({this.workingDirectory, this.environment,
+      this.includeParentEnvironment: true, this.runInShell: false,
+      this.stdoutEncoding: SYSTEM_ENCODING,
+      this.stderrEncoding: SYSTEM_ENCODING});
+
+  /// Create a clone with updated values in one step.
+  /// For omitted parameters values of the original instance are copied.
+  RunOptions clone({
+      String workingDirectory,
+      Map<String,String> environment,
+      bool includeParentEnvironment,
+      bool runInShell,
+      Encoding stdoutEncoding,
+      Encoding stderrEncoding}) {
+    Map<String,String> env;
+    if(environment != null) {
+       env = new Map.from(environment);
+    } else {
+      env = this.environment != null ? new Map.from(this.environment) : {};
+    }
+    return new RunOptions(
+        workingDirectory: workingDirectory != null
+            ? workingDirectory : this.workingDirectory,
+        environment: env,
+        includeParentEnvironment: includeParentEnvironment != null
+            ? includeParentEnvironment : this.includeParentEnvironment,
+        runInShell: runInShell != null
+            ? runInShell : this.runInShell,
+        stdoutEncoding: stdoutEncoding != null
+            ? stdoutEncoding : this.stdoutEncoding,
+        stderrEncoding: stderrEncoding != null
+            ? stderrEncoding : this.stderrEncoding);
+  }
+}
+
+/// Helper for methods which support the deprecated [workingDirectory] and the
+/// new [runOptions] to create a [RunOptions] instance which contains the values
+/// of the passed [runOptions] and the passed [workingDirectory].
+/// If both [workingDirectory] and [runOptions.workingDirectory] are passed
+/// an AssertionError is thrown. Only one of both may be used at one
+/// time.
+/// This function can probably be removed when the deprecated `workingDirectory`
+/// arguments are finally removed.
+RunOptions mergeRunOptions(String workingDirectory, RunOptions runOptions) {
```

Comment:
they are used from two different libraries. I had them private first but then changed it to public. From what I've seen this lib isn't exported from the libs in `lib` anyway therefore it's at least private by convention.